src/runtime_src/core/edge/common/aie_parser.cpp

```diff
@@ -122,6 +123,24 @@ get_rtp(const pt::ptree& aie_meta)
   return rtps;
 }
 
+std::vector<gmio_type>
+get_gmio(const pt::ptree& aie_meta)
+{
+  std::vector<gmio_type> gmios;
+
+  for (auto& gmio_node : aie_meta.get_child("aie_metadata.GMIOs")) {
+    gmio_type gmio;
+
+    gmio.id = gmio_node.second.get<std::string>("id");
+    gmio.name = gmio_node.second.get<std::string>("name");
+    gmio.type = gmio_node.second.get<uint16_t>("type");
+    gmio.shim_col = gmio_node.second.get<uint16_t>("shim_column");
+    gmio.burst_len = gmio_node.second.get<uint16_t>("burst_length_in_16byte");
```

Comment:
Did we miss capturing gmio object into the gmios vector?
`gmios.push_back(gmio)`

Reply:
You are right. I just added. Apparently I am not able to test it without a valid metadata file. Good catch!